wagtail/utils/utils.py

```diff
@@ -40,3 +44,49 @@ def flatten_choices(choices):
             # choice (key, display value)
             ret[str(key)] = value
     return ret
+
+
+def reduce_image_size(avatar, size_bound, max_dimensions=(400, 400)):
+    """
+    Reduce an image's file size to a size_bound in kilobytes.
+
+    :param avatar: The original image gotten from the image field of the form
```

Comment:
If we're putting this function into wagtail/utils/utils.py, that implies that we intend for it to be reusable in multiple places - otherwise we're just splitting up the code for no real benefit. And if this code is indeed meant to be generally reusable, there needs to be more thought about _how_ it's going to be reused - it shouldn't be referring to avatars or "the form" (which form?), and the docstring should be clearer about the input and output data types - in this case it seems to be `UploadedFile`, but perhaps something like [django.core.files.File](https://docs.djangoproject.com/en/5.1/ref/files/file/#django.core.files.File) would ensure that it's not unnecessarily tied to upload forms?

However, if we simplify this logic as suggested below, I suspect we don't need this to be a utility function at all.

Reply:
The logic for its implementation seems to be reusable and seems like a valid use case that might re-occur a number of times, like probably when a user needs an image to be reduced. I had it use UploadedFile due to the fact the field in question is an ImageField moreover the function is mainly for images which should be processed inmemory using the InMemoryUploadFile. 

But i guess you better understand the philosophy of the codebase, i can make the names and docstring more generic for reuse, but if you do think there's no potential usecase it might be reusable for then i can focus it more to the current implementation.